model-engine/model_engine_server/inference/vllm/vllm_server.py

```diff
@@ -77,57 +77,57 @@ async def generate(request: Request) -> Response:
+            background_tasks.add_task(abort_request)
+            return StreamingResponse(stream_results(), background=background_tasks)
+
+        # Non-streaming case
+        final_output = None
+        tokens = []
         last_output_text = ""
         async for request_output in results_generator:
-            log_probs = format_logprobs(request_output)
-            ret = {
-                "text": request_output.outputs[-1].text[len(last_output_text) :],
-                "count_prompt_tokens": len(request_output.prompt_token_ids),
-                "count_output_tokens": len(request_output.outputs[0].token_ids),
-                "log_probs": log_probs[-1] if log_probs and sampling_params.logprobs else None,
-                "finished": request_output.finished,
-            }
+            tokens.append(request_output.outputs[-1].text[len(last_output_text) :])
             last_output_text = request_output.outputs[-1].text
-            yield f"data:{json.dumps(ret)}\n\n"
-
-    async def abort_request() -> None:
-        await engine.abort(request_id)
-
-    if stream:
-        background_tasks = BackgroundTasks()
-        # Abort the request if the client disconnects.
-        background_tasks.add_task(abort_request)
-        return StreamingResponse(stream_results(), background=background_tasks)
-
-    # Non-streaming case
-    final_output = None
-    tokens = []
-    last_output_text = ""
-    async for request_output in results_generator:
-        tokens.append(request_output.outputs[-1].text[len(last_output_text) :])
-        last_output_text = request_output.outputs[-1].text
-        if await request.is_disconnected():
-            # Abort the request if the client disconnects.
-            await engine.abort(request_id)
-            return Response(status_code=499)
-        final_output = request_output
-
-    assert final_output is not None
-    prompt = final_output.prompt
-    ret = {
-        "text": final_output.outputs[0].text,
-        "count_prompt_tokens": len(final_output.prompt_token_ids),
-        "count_output_tokens": len(final_output.outputs[0].token_ids),
-        "log_probs": format_logprobs(final_output),
-        "tokens": tokens,
-    }
-    return Response(content=json.dumps(ret))
+            if await request.is_disconnected():
+                # Abort the request if the client disconnects.
+                await engine.abort(request_id)
+                return Response(status_code=499)
+            final_output = request_output
+
+        assert final_output is not None
+        prompt = final_output.prompt
+        ret = {
+            "text": final_output.outputs[0].text,
+            "count_prompt_tokens": len(final_output.prompt_token_ids),
+            "count_output_tokens": len(final_output.outputs[0].token_ids),
+            "log_probs": format_logprobs(final_output),
+            "tokens": tokens,
+        }
+        return Response(content=json.dumps(ret))
+    except AsyncEngineDeadError as e:
```

Comment:
@anant-marur i'm not sure if this would capture the streaming case. any suggestions?